app/shared/src/androidUnitTest/kotlin/ui/settings/tabs/media/AndroidTorrentCacheViewModelTest.kt

```diff
@@ -0,0 +1,244 @@
+
+        val vm = AndroidTorrentCacheViewModel(context, mediaCacheSettings, permissionManager)
+        vm.refreshStorageState()
+        takeSnapshot()
+
+        assertEquals(
+            AndroidTorrentCacheLocation.ExternalPrivate(
+                File(EXTERNAL_PRIVATE_BASE, DEFAULT_TORRENT_CACHE_DIR_NAME).absolutePath,
+            ),
+            vm.currentSelection,
+        )
+
+        val locationPresentation = vm.torrentLocationPresentation
+        assertEquals(3, locationPresentation.size)
+        assertNotNull(locationPresentation.find { it.value is AndroidTorrentCacheLocation.InternalPrivate })
+        assertNotNull(
+            locationPresentation.find {
+                it.value is AndroidTorrentCacheLocation.ExternalPrivate && it.enabled
+            },
+        )
+    }
+
+    @Test
+    fun `test media cache settings is external private with external unavailable`() = runComposeStateTest {
+        val context = createMockContext(externalAvailable = false)
+        val permissionManager = createTestPermissionManager(null)
+        val mediaCacheSettings by createMockMediaCacheSettingsDelegation(
+            File(EXTERNAL_PRIVATE_BASE, DEFAULT_TORRENT_CACHE_DIR_NAME).absolutePath,
+        )
+        assertTrue { mediaCacheSettings.value.saveDir?.startsWith(EXTERNAL_PRIVATE_BASE) == true }
+
+        val vm = AndroidTorrentCacheViewModel(context, mediaCacheSettings, permissionManager)
+        vm.refreshStorageState()
+        takeSnapshot()
+
+        val locationPresentation = vm.torrentLocationPresentation
+        assertEquals(3, locationPresentation.size)
+        assertNotNull(locationPresentation.find { it.value is AndroidTorrentCacheLocation.InternalPrivate })
+        assertNotNull(
+            locationPresentation.find {
+                it.value is AndroidTorrentCacheLocation.ExternalPrivate && !it.enabled
+            },
+        )
+    }
+
+    @Test
+    fun `test unknown media cache settings`() = runComposeStateTest {
+        val context = createMockContext(externalAvailable = true)
+        val permissionManager = createTestPermissionManager(null)
+        val mediaCacheSettings by createMockMediaCacheSettingsDelegation(
+            File("/system/app/unavailable_path").absolutePath,
+        )
+
+        val vm = AndroidTorrentCacheViewModel(context, mediaCacheSettings, permissionManager)
+        vm.refreshStorageState()
+        takeSnapshot()
+
+        // 如果当前 external private 不可用，那就可能是 external shared。并且同样不可用
+        assertEquals(null, vm.currentSelection)
+
+        val locationPresentation = vm.torrentLocationPresentation
+        assertEquals(3, locationPresentation.size)
+        assertNotNull(locationPresentation.find { it.value is AndroidTorrentCacheLocation.InternalPrivate })
+        assertNotNull(
+            locationPresentation.find {
+                it.value is AndroidTorrentCacheLocation.ExternalPrivate && it.enabled
+            },
+        )
```

Comment:
```suggestion
        vm.torrentLocationPresentation.run {
            assertEquals(3, size)
            assertEquals(..., get(0)
            get 1
            get 2
        }
```

你这样 find 是不够 strict 的